sc-tools/sc-builder/CMakeLists.txt

```diff
@@ -3,15 +3,20 @@ configure_file(
     "${CMAKE_CURRENT_LIST_DIR}/src/builder_defines.hpp"
 )
 
-file(GLOB_RECURSE SOURCES "src/*.cpp" "src/*.h" "src/*.hpp")
+file(GLOB SOURCES CONFIGURE_DEPENDS "src/*.cpp" "src/*.hpp")
 list(REMOVE_ITEM SOURCES "${SC_BUILDER_SRC}/src/main.cpp")
 
 add_library(sc-builder-lib SHARED ${SOURCES})
-include_directories(${SC_MEMORY_SRC} ${SC_CONFIG_UTILS_SRC} ${GLIB2_INCLUDE_DIRS})
-target_link_libraries(sc-builder-lib sc-memory sc-config-utils)
+target_link_libraries(sc-builder-lib
+    LINK_PRIVATE sc-memory
+    LINK_PUBLIC sc-config-utils)
+target_include_directories(sc-builder-lib
+    PRIVATE ${SC_MEMORY_SRC}
+    PUBLIC ${SC_CONFIG_UTILS_SRC}
+)
```

Comment:
Actually I don't get why do we include ${SC_MEMORY_SRC} if all public includes should be propagated why target_link_libraries. Same with config utils - this line should not be needed, nor the variables that are set from other targets